src/server/routes/api.ts

```diff
@@ -0,0 +1,317 @@
+  const files = await FileModel.find({ user });
+  if (!files.length) {
+    res.json([]);
+    return;
+  }
+  const zip = new JSZip();
+  const folder = zip.folder(folderName);
+  for (let i = 0; i < files.length; i++) {
+    if (files[i].path?.indexOf(path) === 0 && files[i].isDeleted !== true) {
+      const fileData = Buffer.from(files[i].data!.buffer);
+      folder!.file(files[i].path!.slice(path.length), fileData);
+    }
+  }
+  const zipped = await zip.generateAsync({ type: "nodebuffer" });
+  const newFile = new FileModel({
+    filename: folderName + ".zip",
+    data: zipped,
+    user: user,
+    size: zipped.length,
+    path: parent + folderName + ".zip",
+  });
+  await newFile.save();
+  res.send("Compress the file successfully");
+});
+
+router.get("/file/download/:id", async (req, res) => {
+  const { id } = req.params;
+  const file = await FileModel.findById(id);
+
+  if (!file) {
+    return res.status(404).send("File not found");
+  }
+
+  res.json({ data: file.data });
+});
+
+router.post("/folder/download", async (req, res) => {
+  const { user, path, folderName, parent } = req.body;
+  const files = await FileModel.find({ user });
+  if (!files.length) {
+    res.json([]);
+    return;
+  }
+  const zip = new JSZip();
+  const folder = zip.folder(folderName);
+  for (let i = 0; i < files.length; i++) {
+    if (files[i].path?.indexOf(path) === 0 && files[i].isDeleted !== true) {
+      const fileData = Buffer.from(files[i].data.buffer);
+      folder!.file(files[i].path.slice(path.length), fileData);
+    }
+  }
+  const zipped = await zip.generateAsync({ type: "nodebuffer" });
+  res.json({
+    filename: folderName + ".zip",
+    data: zipped,
+  });
+});
+
+router.get("/file/detail/:id", async (req, res) => {
+  const { id } = req.params;
+  const file = await FileModel.findById(id);
+
+  if (!file) {
+    return res.status(404).send("File not found");
+  }
+
+  res.json({ data: file.data, filename: file.filename });
+});
+
+router.post("/file/rename/:id", async (req: Request, res: Response) => {
+  const { id } = req.params;
+  const { newname, parent } = req.body;
+  const file = await FileModel.findById(id);
+  if (file) {
+    const paths = file.path?.split("/");
+    file.filename = newname;
+    file.path = parent + newname;
+    await file.save();
+  }
+  res.send("Rename the file successfully");
+});
+
+router.post("/folder/rename", async (req: Request, res: Response) => {
+  const { user, path, newname, parent } = req.body;
+  const files = await FileModel.find({ user });
+  if (!files.length) {
+    res.json([]);
+    return;
+  }
+
+  for (let i = 0; i < files.length; i++) {
+    if (files[i].path?.indexOf(path) === 0 && files[i].isDeleted !== true) {
+      files[i].filename = newname;
+      files[i].path =
+        parent + newname + "/" + files[i].path?.slice(path.length);
+      await files[i].save();
+    }
+  }
+  res.send("Rename the fodler successfully");
```

Comment:
You need to keep consistency to return JSON data as API response.
Also there's a typo in the message.